pyccel/codegen/printing/cwrappercode.py

```diff
@@ -405,7 +409,11 @@ def _body_array(self, variable, collect_var, check_type = False) :
                 error = PyErr_SetString('PyExc_NotImplementedError',
                         '"Argument does not have the expected ordering ({})"'.format(collect_var.order))
                 body += [(PyccelNot(check), [error, Return([Nil()])])]
-        body += [(LiteralTrue(), [Assign(VariableAddress(variable),
+        if self._target_language == "c":
+            cast_function = self.get_cast_function_call('pyccelPyArrayObject_to_ndarray', collect_var)
+            body += [(LiteralTrue(), [Assign(VariableAddress(variable), cast_function)])]
+        else:
+            body += [(LiteralTrue(), [Assign(VariableAddress(variable),
```

Comment:
I'm not sure why this if is necessary here. Isn't this collecting the cast function? Is that not what `get_collect_function_call` is designed to do?